src/rust/engine/process_execution/src/lib.rs

```diff
@@ -584,6 +584,8 @@ pub struct Process {
   pub execution_environment: ProcessExecutionEnvironment,
 
   pub remote_cache_speculation_delay: std::time::Duration,
+
+  pub attempt: usize,
```

Comment:
A comment here explaining what this is used for would be good (sorry for the bad examples from the last few added fields), and that it is only included in `hash`/`eq`, but not the fingerprint.